File: services/recommendation-service/src/recommendation_model.py
from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any

@dataclass
class RecommendationInput:
    profile: dict[str, Any]
    allowed_evidence: list[dict[str, Any]]
    campus: str | None = None
    role: str = "parent"
    consultation_stage: str = "NEEDS_ASSESSMENT"

@dataclass
class ClassRecommendation:
    recommended_class_type: str | None = None
    confidence: str = "low"  # high / medium / low
    reasons: list[str] = field(default_factory=list)
    not_suitable_if: list[str] = field(default_factory=list)
    missing_info: list[str] = field(default_factory=list)
    next_questions: list[str] = field(default_factory=list)
    risk_warnings: list[str] = field(default_factory=list)
    evidence_ids: list[str] = field(default_factory=list)

File: services/recommendation-service/ts-src/recommendationModel.ts
export interface RecommendationInput {
  profile: Record<string, unknown>;
  allowed_evidence: Record<string, unknown>[];
  campus: string | null;
  role: string;
  consultation_stage: string;
}

export const DEFAULT_RECOMMENDATION_INPUT: Omit<RecommendationInput, "profile" | "allowed_evidence"> = {
  campus: null,
  role: "parent",
  consultation_stage: "NEEDS_ASSESSMENT",
};

export interface ClassRecommendation {
  recommended_class_type: string | null;
  confidence: string;
  reasons: string[];
  not_suitable_if: string[];
  missing_info: string[];
  next_questions: string[];
  risk_warnings: string[];
  evidence_ids: string[];
}

export function createClassRecommendation(
  overrides: Partial<ClassRecommendation> = {},
): ClassRecommendation {
  return {
    recommended_class_type: null,
    confidence: "low",
    reasons: [],
    not_suitable_if: [],
    missing_info: [],
    next_questions: [],
    risk_warnings: [],
    evidence_ids: [],
    ...overrides,
  };
}
